src/node_modules.cc

```diff
@@ -498,6 +499,74 @@ void GetCompileCacheDir(const FunctionCallbackInfo<Value>& args) {
           .ToLocalChecked());
 }
 
+void GetCompileCacheEntry(const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args.GetIsolate();
+  CHECK(args[0]->IsString());  // TODO(joyeecheung): accept buffer.
+  CHECK(args[1]->IsString());
+  CHECK(args[2]->IsUint32());
+  Local<Context> context = isolate->GetCurrentContext();
+  Environment* env = Environment::GetCurrent(context);
```

Comment:
```suggestion
  Environment* env = Environment::GetCurrent(args);
  CHECK(args[0]->IsString());  // TODO(joyeecheung): accept buffer.
  CHECK(args[1]->IsString());
  CHECK(args[2]->IsUint32());
```

Then access `env->isolate()` and `env->context()` ... I know we're not super consistent on this but would be nice to work towards more consistency

Reply:
If we are going to have more consistency, we should move _away_ from using `env->context()` as that makes the binding unusable for e.g. shadow realms, and prefer to do what's done here (i.e. use the current context, which makes is usable for shadow realms).

Reply:
Are these types of issues documented somewhere? Style guide updated? Shadow realm related concerns documented?

Reply:
I don't think "the right way to get context" is specifically documented, though it should be inferrable from the notes added in https://github.com/nodejs/node/pull/47932